build/m4/brotli.mk

```diff
@@ -37,4 +39,7 @@ git-brotli:
 	cd lib/brotli && \
 	git reset --hard && \
 	git clean -dfx && \
-	git pull origin master
+	git fetch origin && \
+	git checkout $(BROTLI_GIT_VERSION) && \
+	git submodule sync --recursive && \
+	git submodule update --init --recursive
```

Comment:
I dislike how this requires duplicating the submodule version information which is already stored in our root git itself.
Instead the following can do the same without duplication:
```sh
    cd lib/brotli && \
    git reset --hard && \
    git clean -dfx
    git submodule update --force lib/
```

On another note; those targets seem to have originally been intended to *update* our submodules all at once as the `git-update` name of the target invoking all other `git-*` targets suggests. This commit changes their purpose to resetting/syncing the working state to the HEAD state. Imho this is new purpose is more useful, but perhaps we should then also rename the `git-update` target to something like `git-smreset` or so?